### PR TITLE
Qt build script update

### PR DESCRIPTION
**Pull request by terrychan999**
I've added changes to the Qt build script:
1. bump version to `6.5.0 LTS`
2. add configure options before build
3. replace `make` to `ninja-build`
4. skip extraction if the source code is already extracted
5. add a patch file to remove symlink check

### CODE REVIEW

File: scripts/build.d/qt [hunk 1]
```diff
@@ -39,14 +55,70 @@ pushd ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src/${pkgfull}
...
+  cmakecmd+=("--parallel")
+
+  installcmd=("cmake")
+  installcmd+=("--install")
+  installcmd+=(".")
 
+  buildcmd configure.log "${cfgcmd[@]}"
   buildcmd cmake.log "${cmakecmd[@]}"
-  buildcmd make.log make install -j $NP
+  buildcmd install.log "${installcmd[@]}"
```
Comment (yungyuc) [inline scripts/build.d/qt:120]: I like the change to use `cmake` command to build

**Review by yungyuc — APPROVED** [review]
LGTM

File: scripts/build.d/qt [hunk 2]
```diff
@@ -4,33 +4,49 @@ set -e
...
+  pkgsrc=${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src
+  #pkgsrc=/tmp
+
   download_md5 $pkgfn $pkgurl $pkgmd5
 
-  mkdir -p ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src
-  pushd ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src > /dev/null
-    tar Jxf ${DEVENVDLROOT}/$pkgfn
+  mkdir -p $pkgsrc
+  if [ -d $pkgsrc/$pkgfull ];
```
Comment (yungyuc) [inline scripts/build.d/qt:23]: I missed this.  It behaves differently from other build scripts.  Please add an environment variable to control the behavior.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: scripts/build.d/qt (hunk 2 above)
Comment (yungyuc) [reply scripts/build.d/qt:23]: For example, https://github.com/solvcon/devenv/blob/8e7898bf43a379cb672ec9fd7d061119b3fe8ad3/scripts/build.d/python#L16

File: scripts/build.d/qt [hunk 3]
```diff
@@ -16,21 +25,34 @@ if [ -z "${SYNCGIT}" ]; then
 
   download_md5 $pkgfn $pkgurl $pkgmd5
 
-  mkdir -p ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src
-  pushd ${DEVENVFLAVORROOT}/${DEVENVFLAVOR}/src > /dev/null
-    tar Jxf ${DEVENVDLROOT}/$pkgfn
+  mkdir -p ${QTSRC}
+  if [ -d ${QTSRC}/$pkgfull ];
+  then
+    echo "QTSRC: ${QTSRC}/$pkgfull already exist, skip extract"
```
Comment (yungyuc) [inline scripts/build.d/qt:31]: I want an environment variable to control this behavior.  It should work like `SYNCGIT`.  Users can set the environment variable from outside to skip or force the extract.

**Review by yungyuc — CHANGES_REQUESTED** [review]
@terrychan999 You misunderstood what I meant for the use of the requested environment variable.  I am clarifying in the comment.

File: scripts/build.d/qt (hunk 3 above)
Comment (terrychan999) [reply scripts/build.d/qt:31]: I see... I'll add the controll asap.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**